src/frontends/pytorch/src/op/unfold.cpp

```diff
@@ -0,0 +1,85 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/frontend/pytorch/node_context.hpp"
+#include "openvino/opsets/opset10.hpp"
+
+using namespace ov::opset10;
+
+namespace ov {
+namespace frontend {
+namespace pytorch {
+namespace op {
+
+OutputVector translate_unfold(NodeContext& context) {
+    // constants
+    auto const_0 = context.mark_node(Constant::create(element::i64, Shape{}, {0}));
+    auto const_1 = context.mark_node(Constant::create(element::i64, Shape{}, {1}));
+    auto const_0_list = context.mark_node(Constant::create(element::i64, Shape{1}, {0}));
+    auto const_1_list = context.mark_node(Constant::create(element::i64, Shape{1}, {1}));
+    auto const_neg_1_list = context.mark_node(Constant::create(element::i64, Shape{1}, {-1}));
+
+    // get inputs and prepare auxiliary nodes
+    auto input = context.get_input(0);
+    auto input_shape = context.mark_node(std::make_shared<ShapeOf>(input));
+    auto input_rank = context.mark_node(std::make_shared<ShapeOf>(input_shape));
+
+    auto dimension = context.mark_node(Constant::create(element::i64, Shape{1}, {context.const_input<int64_t>(1)}));
```

Comment:
```suggestion
    auto dimension = context.get_input(1);
```
Since you anyway create `Constant` node there is no reason to convert it to value first. The code I suggested will also work if 1st input is not a constant for example, also it reduces complexity. The only problem may be the shape of existing constant, it can be scalar, if so I still think the best idea would be to add `Unsqueeze` after.

Reply:
Thanks, done.